src/main/java/org/apache/cassandra/distributed/api/IInstance.java

```diff
@@ -23,9 +23,15 @@
 import java.util.UUID;
 import java.util.concurrent.Future;
 
+import org.apache.cassandra.distributed.shared.VersionedApplicationState;
+
 // The cross-version API requires that an Instance has a constructor signature of (IInstanceConfig, ClassLoader)
 public interface IInstance extends IIsolatedExecutor
 {
+    enum StartupOption
```

Comment:
If we leave this in, I would recommend the following

```
interface StartupOption {}
enum StandardStartupOptions { SKIP_RING_JOIN }
```

this enables us to make forward progress without requiring a new release of the interface every time we need a new option.